src/iocore/cache/P_CacheVol.h

```diff
@@ -73,34 +66,7 @@ struct DiskStripe;
 struct CacheVol;
 class CacheEvacuateDocVC;
 
-// Key and Earliest key for each fragment that needs to be evacuated
-struct EvacuationKey {
-  SLink<EvacuationKey> link;
-  CryptoHash           key;
-  CryptoHash           earliest_key;
-};
-
-struct EvacuationBlock {
-  union {
-    unsigned int init;
-    struct {
-      unsigned int done          : 1; // has been evacuated
-      unsigned int pinned        : 1; // check pinning timeout
-      unsigned int evacuate_head : 1; // check pinning timeout
-      unsigned int unused        : 29;
-    } f;
-  };
-
-  int readers;
-  Dir dir;
-  Dir new_dir;
-  // we need to have a list of evacuationkeys because of collision.
-  EvacuationKey       evac_frags;
-  CacheEvacuateDocVC *earliest_evacuator;
-  LINK(EvacuationBlock, link);
-};
-
-class StripeSM : public Continuation, public Stripe
+class StripeSM : public Continuation, public Stripe, public PreservationTable
```

Comment:
Does StripeSM need to inherit PreservationTable or can it be aggregate?   It seems like the few public members could be either internalized or proxied.  I prefer more hasA to isA, especially since StripeSM is already spread across multiple classes.

Reply:
I don't think it's presently an issue for it to be `isA`, but I'm also thinking it should be `hasA`. My reason was that I want to add a constructor, but bringing in some logic from outside `StateSM` is another great reason. The question is whether to do it immediately in this PR. I didn't do it yet because the added delegation is more complicated than doing the superclass and minimizing the changes minimizes risk, but I think it'd be fine to do it now if you think that's better.

Reply:
I don't feel strongly about it, but I do think it would help clean up some boundaries.  Leaving the more impactful refactoring to another PR sounds smart to me as well.